all my own work after my review (please review for inclusion)

ArneBab opened the pull request:
This pull request tracks all the work I did after I finished the review of the original Sharewiki sourcecode. This needs to be reviewed by someone else but me for inclusion as official plugin and for basic quality assurance and for ensuring that no one can get code into Freenet by putting pressure on me.

It would be great if you could check it!

Here’s the associated bug report: https://bugs.freenetproject.org/view.php?id=6519

Commit:
add basic syntax help

Commit:
update l10n

Commit:
add syntax help {toc}

Commit:
import patch

Commit:
more syntax help

Commit:
better code help and blockquote help

Commit:
help: add strong again

Commit:
hopefully really breaking block

Commit:
only redirect to edit if not preview clicked.

Commit:
rename ChangeLog to NEWS, because it is user-facing, not developer facing (as by the GNU coding guidelines).

Commit:
prepared version 0.2.1

Commit:
add Image syntax help.

Commit:
include the NEWS in the source tarball.

Commit:
rename plugin to Sharesite

Commit:
keep the backup format the same: loads ShareWiki backups

Commit:
update version to 0.2.2 with realVersion 2

Commit:
update NEWS

Commit:
actually move to Sharesite

ant seems not to be case sensitive, otherwise this would have been an error.

Commit:
replace freesite URL

Commit:
link to the Sharesite Freesite instead of ShareLink

Commit:
template: Add ids for header and footer to simplify CSS design

Commit:
more consistent IDs for CSS

Commit:
harmonize coding style

Commit:
harmonize coding style

Commit:
added an activelinkUri field.

Commit:
todo: discuss using a redirect for the activelink

is it ok to make it easy to link to files in other containers?

Commit:
increment version to 0.2.4

Commit:
Added tag 0.2.4 for changeset f6689b03c70c

Commit:
Added signature for changeset be98370f3504

Commit:
increment build number to 27

Commit:
add NEWS entry for 0.2.4

Commit:
Added signature for changeset 03a85d80dfae

Commit:
fix: compare strings with .equals()

Commit:
update version and build number.

Commit:
Added tag 0.2.5 for changeset 4489ccf64d09

Commit:
Added signature for changeset 038bde8cbc27

Commit:
improve news

Commit:
activelink: note that the URI must be SSK or CHK

Commit:
prepare 0.2.6

Commit:
help: {toc} only works as only element in the line

Commit:
mention that the h. headline tags are what are used to form the contents text

thanks to baffled!

Commit:
syntax: note that to the left is the syntax

Commit:
add a very basic Makefile

Commit:
prepare version 0.2.7

Commit:
fix Makefile a bit and update build number.

Commit:
Added tag 0.2.7 for changeset 06c3625ff20b

Commit:
Added signature for changeset cd12c2e21dad

Commit:
Added signature for changeset e3c31f3768f3

Thynix commented:
5ae01867ba82f562215b56baa84276250a33dbbc looks good to me.

Commit:
syntax help, block elem first plus tabify

Commit:
parse description as textile and name content content. Thanks to banbananabans@CQUZ... from FMS

Commit:
also link to style.css at the bottom

Commit:
Add explicit path option

Commit:
also use the path for the link in the inserter and the site list

Commit:
footer: link to the latest version of Sharesite

Commit:
Update NEWS file

Commit:
ensure that getPath always has something to return

Commit:
url encode / in paths, because it breaks preview and insert.

Commit:
template: enclose description by div, not by p

Commit:
allow restricting inserts to a certain hour of the day

Commit:
improve UI strings

Commit:
fix source selection in Makefile

Commit:
allow setting the upload time to a random hour

Commit:
FIX: adjusted just the parameter, not the attribute

Commit:
Added signature for changeset 76c0e45f2bee

Commit:
Added tag 0.2.8 for changeset efb842bd4f73

Commit:
Added signature for changeset 7f938354de9c

Commit:
merge git tag and hg sign

Commit:
update NEWS

Commit:
add TODO for a random wait time.

Commit:
Makefile: Always run ant clean before ant dist

Commit:
get the version of the running plugin instance, not the plugin class.

Commit:
add a random wait time within the selected hour interval to mask triggering an insert in that hour

Commit:
public static final -> private static final *happy*

Commit:
Increase version to 0.4.0: We now support convenient anonymous usage

Commit:
update NEWS

Commit:
textilej: add a special case for freenet keys (thank you, you know who you are)

Commit:
file bugs to keep the feature requests from FMS

Commit:
add a template for the initial content (preparation to ease starting to write)

Commit:
show insertHour setting after the path (at the top)

Commit:
version 0.4.1

Commit:
insertHour below content field

Commit:
add default content

Commit:
update NEWS

Commit:
Added tag 0.4.1 for changeset 2a59727908b1

Commit:
Added signature for changeset 7d4d68e90ef7

Bombe commented on src/plugins/Sharesite/Freesite.java:
```diff
@@ -76,7 +76,22 @@ public Freesite(int uniqueKey) {
 		}
 
+		try {
+			InputStream is = Plugin.class.getClassLoader().getResourceAsStream(texttemplate);
```
Potential resource leak here. Check `try-with-resources` to prevent that.

ArneBab replied:
Thank you for the catch! I now switched do try-with-resources for both the text loading and the css loading for both the input stream and the BufferedReader.

Commit:
switch to Java 1.7 as source and target

Commit:
use try-with-resources to avoid resource leaks. Thanks to Bombe\!

Commit:
cleanup whitespace

ArneBab commented:
@Bombe does your comment mean that you reviewed the changes so we could deploy the new version once I can create proper releases? (that would be awesome!)

Bombe commented:
@ArneBab, I’m pretty sure that I did not review it in any depth because I would totally, completely, utterly, forcefully reject using a `Makefile` to compile Java software.

ArneBab commented:
Maybe I should add a note to the Makefile that it exists to make Emacs flymake work out of the box.

/me resists the urge to start a build environment war and sucks all the comments from his tongue ☺

Commit:
add show current utc time bug

Commit:
Show the current UTC hour next to the InsertHour field

Commit:
FIX: used wrong replacement value for current utc hour

Bombe commented on src/plugins/Sharesite/webui/EditToadlet.java:
```diff
@@ -128,9 +131,14 @@ private void addNodes( HTMLNode form, String name, String path, String desc, Str
+		Integer currentHour = Calendar.getInstance(TimeZone.getTimeZone("UTC"), Locale.US)
+				.get(Calendar.HOUR_OF_DAY); // 0-23
+		Integer currentMinute = Calendar.getInstance(TimeZone.getTimeZone("UTC"), Locale.US)
+				.get(Calendar.MINUTE); // 0-59
```
* I don’t see you using the `currentMinute` anywhere.
* You should use a single `Calendar` instance for getting the two values.

ArneBab replied:
Ah, yes, I stopped using it in the L10N (because I did not see how to pass it two values) but forgot to remove the instance. Thank you!

Commit:
update to 0.4.2

Commit:
remove unused variable, thanks to Bombe

Commit:
Added tag 0.4.2 for changeset df61855bf1ad

Commit:
Added signature for changeset 7e2afe9b55ca

ArneBab committed:
increase realVersion and version

Commit:
ant: add a copy task to provide a Sharesite.jar for the release scripts

ArneBab committed:
0.4.4 with updated ant

Commit:
fix regression: busy wait for queued websites

Commit:
Added tag 0.4.5 for changeset 792153e157d9

Commit:
Added signature for changeset 63ac7f7777df

Commit:
Makefile: avoid rebuilding without changes

Commit:
Makefile: check and check-syntax dependencies

They should depend on SOURCES, not just the state of src/

Commit:
add missing checks for null values

Commit:
refactor: rename variable iHour to plain hour

Commit:
add bug

Commit:
add syntax help for the audio tag

Commit:
content-template: add a tip against style analysis

Commit:
break run-loop when interrupted

Commit:
Inserter: actually *break* when interrupted

Commit:
Inserter: whitespace cleanup

Commit:
Prepare for 0.4.6

Commit:
Use a ticker instead of a while loop.

Probably doing it wrong, but it seems to work.

Commit:
merge useticker branch

Commit:
tick within 10ms after requesting an insert

Commit:
update version to 0.4.7

Commit:
fix: undid my breaking change to the correct pull-request of TheSeeker

ArneBab committed:
build.xml: adapt to new jarpath of gradle-based fred

ArneBab committed:
more beautiful default style + example for header image using the activelink.

Commit:
new default css thanks to w7Q2J@QTPnftHRW6dAF… from FMS

Readability increases a lot, and it seems more friendly while keeping
it simple.

Commit:
merge
